### PR TITLE
Feat: Encrypt Password

### PR DESCRIPTION
**Pull request by Orivera04**
Description:

In this PR Spring security was integrated, with this integration we will encrypt the password of the users that uses the app.

ScreenShots:
CleanShot 2024-10-31 at 23 24 47

### CODE REVIEW

**Comment by Orivera04** [comment]
@Rennerp If I'm Right, the scope of this task it's only to encrypt passwords right ? 

If that is correct, in the next task, we would need a security mechanism, I was thinking in use JWT tokens, So we need to create a Login Controller that will need a user and a password, if the login data is correct, we will return a JWT token, and we will use that token to authenticate all the request of every controller of the app but LoginController and UserController (If we need to create a new user) I think.

I am on Telegram if you have questions or need some clarification :)

File: src/main/java/com/grapeinc/taskator_server/services/UserService.java [hunk 1]
```diff
@@ -88,6 +94,14 @@ public Optional<User> updatePassword(Long id, Map<String, Object> fields) {
...
 
             if (existingUserOptional.isPresent()) {
+                // Check if the fields map contains the password field
+                // If it does, update the password
+                if (fields.containsKey("password")) {
+                    String rawPassword = (String) fields.remove("password");
+                    existingUserOptional.get().setPassword(passwordEncoder.encode(rawPassword));
+                    fields.remove("password");
+                }
+
```
Comment (Rennerp) [inline src/main/java/com/grapeinc/taskator_server/services/UserService.java:104]: ~~Is this necessary? This method was created specifically for updating the password why are we checking if fields contains it?~~

~~Don't think we will ever this method if its not for updating the password.~~

Got it know, I wonder how can we improve the hashing of the password to be something that goes by default everytime that we are setting the property, Is this something we could achieve?

File: src/main/java/com/grapeinc/taskator_server/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+public class SecurityConfig {
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    // TODO: Implement security configuration
+    // This is a very basic security configuration that disables CSRF and authorizes all requests
+    // We need to define if we want to use a JWT token, OAuth2, or any other security mechanism
+    // So for the moment, we are going to disable CSRF and authorize all requests
```
Comment (Rennerp) [inline src/main/java/com/grapeinc/taskator_server/config/SecurityConfig.java:22]: Can we come up with an assessment on what suits the best for this case?

**Review by Rennerp — APPROVED** [review]
Just few comments nothing to change on the code, LGTM

**Comment by Rennerp** [comment]
> @Rennerp If I'm Right, the scope of this task it's only to encrypt passwords right ?
> 
> If that is correct, in the next task, we would need a security mechanism, I was thinking in use JWT tokens, So we need to create a Login Controller that will need a user and a password, if the login data is correct, we will return a JWT token, and we will use that token to authenticate all the request of every controller of the app but LoginController and UserController (If we need to create a new user) I think.
> 
> I am on Telegram if you have questions or need some clarification :)

All of this ideas sounds good to me, feel free to start working on those! I'm merging this code, one thig that we missed was the PUT request, we are updating the password in that scenario, don't know how it will behave with the hashing of the password. This is something that you can fix as part of your next task.

**Pull request merged**